roofit/roofitcore/src/RooMinimizer.cxx

```diff
@@ -816,8 +799,61 @@ void RooMinimizer::profileStop()
 }
 
 
+ROOT::Math::IMultiGenFunction* RooMinimizer::getFitterMultiGenFcn() const
+{
+   return fitter()->GetFCN();
+}
+
+
+ROOT::Math::IMultiGenFunction* RooMinimizer::getMultiGenFcn() const
+{
+   if (getFitterMultiGenFcn()) {
+      return getFitterMultiGenFcn();
+   } else {
+      switch (_fcnMode) {
+      case FcnMode::classic: {
+         return static_cast<ROOT::Math::IMultiGenFunction *>(static_cast<RooMinimizerFcn *>(_fcn));
+      }
+      case FcnMode::gradient: {
+         auto thing = dynamic_cast<RooGradMinimizerFcn *>(_fcn);
+         if (thing != nullptr) {
+            return static_cast<ROOT::Math::IMultiGenFunction *>(thing);
+         } else {
+            throw std::logic_error("In RooMinimizer::fitterFcn: Minimizer fcnMode was set to gradient, but fit function type does not match RooGradMinimizerFcn!");
+         }
+      }
```

Comment:
You can probably fix the warning like this. The compiler complains that the switch is incomplete:
```suggestion
      }
      default: return nullptr;
```